### PR TITLE
V22F-246 Simulation run time

### PR DESCRIPTION
**Pull request by klieverse**

### CODE REVIEW

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp [hunk 1]
```diff
@@ -251,3 +251,26 @@ void TestBoolArrMediator(bool p_env, bool p_car, bool p_human, bool p_interventi
...
+TEST(MediatorTests, TimeOutTest)
+{
+    Random random;
+    char path[256];
+    for (int j = 0; j < TEST_AMOUNT; j++)
+    {
+        SDAConfigMediator::ClearInstance();
+        ASSERT_TRUE(SetupSingletonsFolder());
+
+        int maxTimeMinutes = random.NextInt();
```
Comment (suppergerrie2) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp:265]: You should use random.NextInt(minimum, maximum) so you do not get gigantic numbers or negative numbers. (Tho a test testing on those would also be good, testing edge cases is good)

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.h [hunk 2]
```diff
@@ -51,6 +52,12 @@ class Mediator
     {
         return &m_decisionMaker;
     }
+
+    void SetTicks(unsigned long tick)
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.h:56]: p_tick

**Review by KungFuDonkey — CHANGES_REQUESTED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp [hunk 3]
```diff
@@ -251,3 +251,26 @@ void TestBoolArrMediator(bool p_env, bool p_car, bool p_human, bool p_interventi
...
+TEST(MediatorTests, TimeOutTest)
+{
+    Random random;
+    char path[256];
+    for (int j = 0; j < TEST_AMOUNT; j++)
+    {
+        SDAConfigMediator::ClearInstance();
+        ASSERT_TRUE(SetupSingletonsFolder());
+
+        int maxTimeMinutes = random.NextInt(0, 1440);
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp:265]: Name a define for 1440

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp [hunk 4]
```diff
@@ -251,3 +251,26 @@ void TestBoolArrMediator(bool p_env, bool p_car, bool p_human, bool p_interventi
...
+        SDAConfigMediator::GetInstance()->GetDecisionMaker()->Config.SetMaxTime(maxTimeMinutes);
+
+        unsigned long currentTick = random.NextUInt(14400000);
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/MediatorTests.cpp:268]: name a define for 1400000

**Review by KungFuDonkey — CHANGES_REQUESTED** [review]

**Review by KungFuDonkey — APPROVED** [review]

**Review by suppergerrie2 — CHANGES_REQUESTED** [review]
Also add the timeout check to the replayDriver/Driver.cpp since otherwise it wont get tested

**Review by suppergerrie2 — APPROVED** [review]

**Review by suppergerrie2 — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp [hunk 5]
```diff
@@ -427,12 +428,26 @@ TEST(RecorderTests, UpgradeFromV0Test)
     ASSERT_FALSE(GfParmExistsParam(upgradedRunSettingsHandle, PATH_PARTICIPANT_CONTROL, KEY_PARTICIPANT_CONTROL_BB_RECORD_SESSION));
 }
 
-TEST(RecorderTests, ValidateV1RecordingTest)
+TEST(RecorderTests, UpgradeFromV1Test)
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:431]: These tests need comments

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp [hunk 6]
```diff
@@ -427,12 +428,26 @@ TEST(RecorderTests, UpgradeFromV0Test)
...
 
+    void* upgradedRunSettingsHandle = GfParmReadFile(filesystem::path(toUpgrade).append(RUN_SETTINGS_FILE_NAME).string().c_str(), 0, true);
+
+    ASSERT_NE(upgradedRunSettingsHandle, nullptr);
+
+    ASSERT_EQ(GfParmGetNum(upgradedRunSettingsHandle, PATH_VERSION, KEY_VERSION, nullptr, NAN), CURRENT_RECORDER_VERSION);
+    ASSERT_EQ(GfParmGetNum(upgradedRunSettingsHandle, PATH_MAX_TIME, KEY_MAX_TIME, nullptr, NAN), DEFAULT_MAX_TIME);
+}
+
+TEST(RecorderTests, ValidateLatestRecordingTest)
```
Comment (KungFuDonkey) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp:445]: These tests need comments

**Review by KungFuDonkey — CHANGES_REQUESTED** [review]

**Review by KungFuDonkey — APPROVED** [review]

File: speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.inl [hunk 7]
```diff
@@ -192,6 +193,7 @@ void Mediator<DecisionMaker>::DriveTick(tCarElt* p_car, tSituation* p_situation)
 template <typename DecisionMaker>
 void Mediator<DecisionMaker>::RaceStart(tTrack* p_track, void* p_carHandle, void** p_carParmHandle, tSituation* p_situation, Recorder* p_recorder)
 {
+    m_tickCount = 0;
```
Comment (noutheesink) [inline speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Mediator.inl:196]: The Mediator and the DecisionMaker now both have their own tickCount, though their values should always be the same. I think it would be better to make a getter in the DecisionMaker, that the Mediator can call

**Review by noutheesink — CHANGES_REQUESTED** [review]

**Review by noutheesink — APPROVED** [review]
